crates/api_models/src/subscription.rs

```diff
@@ -0,0 +1,151 @@
+use common_utils::{events::ApiEventMetric, pii};
+use masking::Secret;
+use utoipa::ToSchema;
+
+use crate::{
+    customers::{CustomerRequest, CustomerResponse},
+    payments::CustomerDetailsResponse,
+};
+
+/// Request payload for creating a subscription.
+///
+/// This struct captures details required to create a subscription,
+/// including plan, profile, merchant connector, and optional customer info.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct CreateSubscriptionRequest {
+    /// Unique identifier for the subscription (optional, generated if missing).
+    pub subscription_id: Option<common_utils::id_type::SubscriptionId>,
```

Comment:
We should not get the database reference id from merchants, id should be generated for us, introduce a merchant_reference_id where they can pass their ids but that is not the id we will be processing with our api's.

The id which we generate should be global unique without any composite id logic